packages/core/src/transport-http.ts

```diff
@@ -551,53 +450,104 @@ export class StreamableHttpTransport {
       "Content-Type": SSE_ACCEPT_HEADER,
       Connection: "keep-alive",
     };
-    if (this.generateSessionId) {
-      const sid = request.headers.get(MCP_SESSION_ID_HEADER);
-      if (sid) headers[MCP_SESSION_ID_HEADER] = sid;
+    if (this.generateSessionId && sessionId) {
+      headers[MCP_SESSION_ID_HEADER] = sessionId;
     }
 
-    return new Response(responseStream, {
+    return new Response(stream as ReadableStream, {
       status: 200,
       headers,
     });
   }
 
-  private async handleDelete(request: Request): Promise<Response> {
-    const sessionId = request.headers.get(MCP_SESSION_ID_HEADER);
+  private async handleGet(request: Request): Promise<Response> {
+    const accept = request.headers.get("Accept");
+    if (!accept || !accept.includes(SSE_ACCEPT_HEADER)) {
+      return new Response(
+        "Bad Request: Accept header must be text/event-stream",
+        {
+          status: 400,
+        },
+      );
+    }
+
+    const protocolHeader = request.headers.get(MCP_PROTOCOL_HEADER);
+    if (protocolHeader && protocolHeader !== SUPPORTED_MCP_PROTOCOL_VERSION) {
+      return new Response("Bad Request: Protocol version mismatch", {
+        status: 400,
+      });
+    }
+
     if (!this.generateSessionId) {
+      // Stateless mode does not provide a standalone GET stream
       return new Response("Method Not Allowed", { status: 405 });
     }
-    if (!sessionId) {
-      return new Response("Bad Request: Missing session ID", {
+
+    const sessionId = request.headers.get(MCP_SESSION_ID_HEADER);
+    if (!sessionId || !(await this.sessionStore?.has(sessionId))) {
```

Comment:
note to self - if session id is not in session store, spec says we should return a 404